crates/ty_python_semantic/src/types/diagnostic.rs

```diff
@@ -1762,21 +1762,29 @@ pub(super) fn report_possibly_unbound_attribute(
     ));
 }
 
-pub(super) fn add_inferred_python_version_hint(db: &dyn Db, mut diagnostic: LintDiagnosticGuard) {
+/// Add a subdiagnostic to `diagnostic` that explains why a certain Python version was inferred.
+///
+/// ty can infer the Python version from various sources, such as command-line arguments,
+/// configuration files, or defaults.
+pub fn add_inferred_python_version_hint_to_diagnostic(
```

Comment:
this function could plausibly be moved to the `ruff_db` crate now, since it's no longer specific to `ty_python_semantic` diagnostics. I'm happy to make that change, but it's a slightly bigger refactor -- the `PythonVersionSource` enum would also have to be moved to that crate

Reply:
I'd prefer to keep it in `ty_python_semantic`. While it's possible dependency-wise, it still more closely fits into a Python-specific crate. 

I do agree that it's no longer typing specific, which could be a reason to move it out of the `types` module.

```suggestion
/// Add a subdiagnostic to `diagnostic` that explains why a certain Python version was inferred.
///
/// ty can infer the Python version from various sources, such as command-line arguments,
/// configuration files, or defaults.
pub(crate) fn add_inferred_python_version_hint_to_diagnostic(
```

Reply:
Making it `pub(crate)` doesn't compile, because this PR adds a usage of this function from the `ty_project` crate. That's part of the reason why I thought it might be logical to move the function to `ruff_db` -- it's a utility function that can be used to enhance multiple kinds of diagnostics, not all of which are emitted or even processed by the `ty_python_semantic` crate.

Reply:
I still think it's a good fit given that ProgramSettings are defined in the semantic crate

Reply:
I realised we already had a `utils` module, so I moved it there ;)